keras/src/export/litert.py

```diff
@@ -0,0 +1,548 @@
+        """
+        self.model = model
+        self.input_signature = input_signature
+        self.verbose = verbose
+        self.aot_compile_targets = aot_compile_targets
+        self.kwargs = kwargs
+
+    def export(self, filepath):
+        """Exports the Keras model to a TFLite file and optionally performs AOT
+        compilation.
+
+        Args:
+            filepath: Output path for the exported model
+
+        Returns:
+            Path to exported model or compiled models if AOT compilation is
+            performed
+        """
+        if self.verbose:
+            io_utils.print_msg("Starting LiteRT export...")
+
+        # 1. Ensure the model is built by calling it if necessary
+        self._ensure_model_built()
+
+        # 2. Resolve / infer input signature
+        if self.input_signature is None:
+            if self.verbose:
+                io_utils.print_msg("Inferring input signature from model.")
+            from keras.src.export.export_utils import get_input_signature
+
+            self.input_signature = get_input_signature(self.model)
+
+        # 3. Convert the model to TFLite.
+        tflite_model = self._convert_to_tflite(self.input_signature)
+
+        if self.verbose:
+            # Calculate model size from the serialized bytes
+            final_size_mb = len(tflite_model) / (1024 * 1024)
+            io_utils.print_msg(
+                f"TFLite model converted successfully. Size: "
+                f"{final_size_mb:.2f} MB"
+            )
+
+        # 4. Save the initial TFLite model to the specified file path.
+        assert filepath.endswith(".tflite"), (
+            "The LiteRT export requires the filepath to end with '.tflite'. "
+            f"Got: {filepath}"
+        )
+
+        with open(filepath, "wb") as f:
+            f.write(tflite_model)
+
+        if self.verbose:
+            io_utils.print_msg(f"TFLite model saved to {filepath}")
+
+        # 5. Perform AOT compilation if targets are specified and LiteRT is
+        # available
+        compiled_models = None
+        if self.aot_compile_targets and litert.available:
+            if self.verbose:
+                io_utils.print_msg(
+                    "Performing AOT compilation for LiteRT targets..."
+                )
+            compiled_models = self._aot_compile(filepath)
+        elif self.aot_compile_targets and not litert.available:
+            logging.warning(
+                "AOT compilation requested but LiteRT is not available. "
+                "Skipping AOT compilation."
+            )
+
+        if self.verbose:
+            io_utils.print_msg(
+                f"LiteRT export completed. Base model: {filepath}"
+            )
+            if compiled_models:
+                io_utils.print_msg(
+                    f"AOT compiled models: {len(compiled_models.models)} "
+                    "variants"
+                )
+
+        return compiled_models if compiled_models else filepath
```

Comment:
It's a bit of weird API. Should we save the compiled models as files instead? In this case, the file names should be passed to __init__.